cpp/include/cudf/types.h

```diff
@@ -85,6 +85,36 @@ typedef struct {
 	// here we can also hold info for decimal datatype or any other datatype that requires additional information
 } gdf_dtype_extra_info;
 
+/**---------------------------------------------------------------------------*
+ * @union gdf_data
```

Comment:
Shouldn't this be `gdf_datum`, signifying the singular rather than the plural?

`gdf_data` sounds super- vague, and is also confusing IMHO since it can be interpreted to mean lots of data, or all data in a column etc.